public/app/core/components/TransformersUI/rowsToFields/rowsToFields.ts

```diff
@@ -0,0 +1,193 @@
+import { map } from 'rxjs/operators';
+import {
+  anyToNumber,
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  DataTransformerInfo,
+  Field,
+  FieldColorModeId,
+  FieldConfig,
+  FieldType,
+  getFieldDisplayName,
+  ReducerID,
+  ThresholdsMode,
+} from '@grafana/data';
+
+export interface RowToFieldsTransformOptions {
+  nameField?: string;
+  valueField?: string;
+  mappings?: RowToFieldsTransformMappings[];
+}
+
+export interface RowToFieldsTransformMappings {
+  fieldName: string;
+  reducerId?: ReducerID;
+  configProperty: string;
+}
+
+export const rowsToFieldsTransformer: DataTransformerInfo<RowToFieldsTransformOptions> = {
+  id: DataTransformerID.rowsToFields,
+  name: 'Rows to fields',
+  description: 'Convert rows to fields with dynamic config',
```

Comment:
Will we ever create multiple fields from a single row?  Is there a good use case for that?   If not, perhaps:

```suggestion
  description: 'Convert each row into a field with dynamic config',
```